src/components/scenes/CreateWalletImportOptionsScene.tsx

```diff
@@ -0,0 +1,232 @@
+import { SceneWrapper } from '../common/SceneWrapper'
+import { CryptoIcon } from '../icons/CryptoIcon'
+import { TextInputModal } from '../modals/TextInputModal'
+import { Airship } from '../services/AirshipInstance'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+import { EdgeText } from '../themed/EdgeText'
+import { MainButton } from '../themed/MainButton'
+import { SceneHeader } from '../themed/SceneHeader'
+import { Tile } from '../tiles/Tile'
+
+interface Props {
+  navigation: NavigationProp<'createWalletImportOptions'>
+  route: RouteProp<'createWalletImportOptions'>
+}
+
+const CreateWalletImportOptionsComponent = (props: Props) => {
+  const { navigation, route } = props
+  const { createWalletList, fiatCode, importText, walletNames } = route.params
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
+  const account = useSelector(state => state.core.account)
+  const { currencyConfig } = account
+
+  const importOpts = React.useMemo<Map<string, Set<ImportKeyOption>>>(() => {
+    const pluginIdMap: Map<string, Set<ImportKeyOption>> = new Map()
+
+    for (const createItem of createWalletList) {
+      const { pluginId, tokenId } = createItem
+      const { importKeyOptions } = SPECIAL_CURRENCY_INFO[pluginId] ?? {}
+      if (importKeyOptions == null || tokenId != null) continue
+
+      if (!pluginIdMap.has(pluginId)) {
+        pluginIdMap.set(pluginId, new Set(importKeyOptions))
+      }
+    }
+
+    return pluginIdMap
+  }, [createWalletList])
+
+  const [values, setValues] = React.useState<Map<string, { value: string; error?: string }>>(() => {
+    const valueMap = new Map()
+    for (const [pluginId, opts] of importOpts.entries()) {
+      opts.forEach(opt => {
+        valueMap.set(getOptionKey(pluginId, opt), { value: '', error: undefined })
+      })
+    }
+    return valueMap
+  })
+
+  const disableNextButton = ![...importOpts.entries()].every(([pluginId, opts]) => {
+    for (const opt of [...opts]) {
+      const key = getOptionKey(pluginId, opt)
+      const input = values.get(key)
+      if (input == null) continue
+
+      if (input.error != null || (input.value === '' && opt.required)) {
+        return false
+      }
+    }
+
+    return true
+  })
+
+  const handleValueChange = useHandler((input: string, pluginId: string, opt: ImportKeyOption) => {
+    const key = getOptionKey(pluginId, opt)
+
+    if (input === '' || opt.inputValidation.test(input)) {
+      setValues(map => new Map(map.set(key, { value: input, error: undefined })))
+    } else {
+      setValues(map => new Map(map.set(key, { value: input, error: 'some string about invalid input' })))
+    }
+  })
+
+  const handleEditValue = (initialValue: string, pluginId: string, opt: ImportKeyOption) => {
+    const onSubmit = async (input: string) => {
+      if (input === '') return true
+      return await currencyConfig[pluginId]
+        .importKey(importText, { keyOptions: { [opt.optionName]: input } })
```

Comment:
I originally had it configured to test the import completely only after tapping next but was concerned about losing context for errors. I settled for the current way because I thought knowing which field was incorrect was better experience but you're right this will break immediately once there are more than one required fields.